src/server/tablet_snap.rs

```diff
@@ -805,6 +804,17 @@ where
             }
             Task::Send { addr, msg, cb } => {
                 let region_id = msg.get_region_id();
+                let to_peer_id = msg.get_to_peer().get_id();
+                defer!({
+                    if let Err(e) = self.snap_mgr.delete_snapshot(region_id, to_peer_id) {
```

Comment:
If the node shutdowns after generating snapshot but before this line, the snapshot may still not be cleared? Or do we have some checks and actions on node start?

Reply:
It will resent snapshot again after this node restart. So the old snapshot will be deleted .

Reply:
If it's not a leader after restart?

Reply:
Good catch , I will remove all `gen_x_x_x_x` file  after restart.